### PR TITLE
Fix rest detection timescale

### PR DESCRIPTION
**Pull request by wigwagwent**
Sensor fusion expects the time to be in seconds while rest detection expects it to be in microseconds. This makes is so when the update sensor fusion is called the rest detection now gets the time in micros. Also linear acceleration was missing parentheses giving wrong results (only lsm6dsv uses this to my knowledge)

### CODE REVIEW

**Comment by ButterscotchV** [comment]
The time scale can probably be moved to `RestDetection.h` can't it? I think it's odd to have it different, otherwise this does look good I think?

File: src/sensors/SensorFusion.cpp [hunk 1]
```diff
@@ -173,9 +173,9 @@ namespace SlimeVR
         
         void SensorFusion::calcLinearAcc(const sensor_real_t accin[3], const sensor_real_t gravVec[3], sensor_real_t accout[3])
         {
-            accout[0] = accin[0] - gravVec[0] * CONST_EARTH_GRAVITY;
-            accout[1] = accin[1] - gravVec[1] * CONST_EARTH_GRAVITY;
-            accout[2] = accin[2] - gravVec[2] * CONST_EARTH_GRAVITY;
+            accout[0] = (accin[0] - gravVec[0]) * CONST_EARTH_GRAVITY;
+            accout[1] = (accin[1] - gravVec[1]) * CONST_EARTH_GRAVITY;
+            accout[2] = (accin[2] - gravVec[2]) * CONST_EARTH_GRAVITY;
```
Comment (0forks) [inline src/sensors/SensorFusion.cpp:178]: Breaking change for every IMU other than BNO.

Vector `accin` is in m/s^2 and `gravVec` is a unit vector, if you have issues with linear acceleration in another IMU implementation, perhaps you're passing acceleration into fusion in G?

**Review by 0forks — CHANGES_REQUESTED** [review]

File: src/sensors/SensorFusion.cpp (hunk 1 above)
Comment (wigwagwent) [reply src/sensors/SensorFusion.cpp]: I did not realize I was using this wrong, I have reverted my changes and will go about this another way in the IMU specific code. I will probably change the units the IMU uses.

**Comment by wigwagwent** [comment]
I changed the rest detection to be in seconds instead of in microseconds like it was before. **I am not able to test this.** The best way to test would be see if you can still calibrate a bmi160.

File: src/motionprocessing/RestDetection.h [hunk 2]
```diff
@@ -20,14 +20,14 @@
 struct RestDetectionParams {
     sensor_real_t biasClip;
     sensor_real_t biasSigmaRest;
-    uint32_t restMinTimeMicros;
+    uint32_t restMinTime;
```
Comment (0forks) [inline src/motionprocessing/RestDetection.h:23]: `uint32_t` -> `sensor_real_t`

File: src/motionprocessing/RestDetection.h [hunk 3]
```diff
@@ -235,7 +235,7 @@ class RestDetection {
 private:
     RestDetectionParams params;
     bool restDetected;
-    uint32_t restTimeMicros;
+    uint32_t restTime;
```
Comment (0forks) [inline src/motionprocessing/RestDetection.h:238]: `uint32_t` -> `sensor_real_t`

File: src/motionprocessing/RestDetection.h [hunk 4]
```diff
@@ -134,13 +134,13 @@ class RestDetection {
...
                 || fabs(restLastGyrLp[2]) > biasClip) {
-            restTimeMicros = 0;
+            restTime = 0;
             restDetected = false;
         }
 #endif
     }
 
-    void updateAcc(uint32_t dtMicros, sensor_real_t acc[3]) {
+    void updateAcc(uint32_t dt, sensor_real_t acc[3]) {
```
Comment (0forks) [inline src/motionprocessing/RestDetection.h:143]: `uint32_t` -> `sensor_real_t`

**Review by 0forks — CHANGES_REQUESTED** [review]
Tested, need to change the types to `sensor_real_t`, otherwise it all gets rounded and deltas become 0.

Also minor inconsistency with tabs/spaces, but we need to fix that across the whole codebase honestly.

**Review by Eirenliel — COMMENTED** [review]
Update from comments

**Review by Eirenliel — APPROVED** [review]

**Pull request merged**